scripts/quests/windurst/SOB4_Onion_Rings.lua

```
@@ -0,0 +1,238 @@
+                    totalHoursLeft = 24 - (VanadielHour() + daysPassed * 24) + quest:getVar(player, "HourStarted")
+                    if totalHoursLeft > 0 then
+                        return quest:event(436)
+                    else
+                        return quest:event(443)
+                    end
+                end,
+            },
+            ['Shanruru'] =
+            {
+                onTrigger = function(player, npc)
+                    daysPassed     = VanadielDayOfTheYear() - quest:getVar(player, "DayStarted")
+                    totalHoursLeft = 24 - (VanadielHour() + daysPassed * 24) + quest:getVar(player, "HourStarted")
+                    if totalHoursLeft > 0 then
+                        return quest:event(439)
+                    else
+                        return quest:event(446)
+                    end
+                end,
+            },
+            ['Yafa_Yaa'] =
+            {
+                onTrigger = function(player, npc)
+                    daysPassed     = VanadielDayOfTheYear() - quest:getVar(player, "DayStarted")
+                    totalHoursLeft = 24 - (VanadielHour() + daysPassed * 24) + quest:getVar(player, "HourStarted")
+                    if totalHoursLeft > 0 then
+                        return quest:event(437)
+                    else
+                        return quest:event(444)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [433] = function(player, csid, option, npc)
+                    if quest:complete(player) then
+                        player:delKeyItem(xi.ki.OLD_RING)
+                        player:needToZone(true)
+                    end
+                end,
+            },
+        },
+
+        [xi.zone.WINDURST_WALLS] =
+        {
+            ['_6n2'] =
+            {
+                onTrigger = function(player, npc)
```

With the example function above, you could do something similar, except pass nil to the second event ID parameter

---

I'll use the regular "nothing to see here" interaction as second param. I think thats better